File: sample/src/main/java/com/github/lany192/box/sample/data/binder/ArticleBinder.java
package com.github.lany192.box.sample.data.binder;

import android.view.View;

import androidx.annotation.NonNull;

import com.alibaba.android.arouter.AppRouter;
import com.github.lany192.arch.adapter.BindingHolder;
import com.github.lany192.arch.items.ItemBinder;
import com.github.lany192.arch.utils.DateUtils;
import com.github.lany192.box.sample.MockUtils;
import com.github.lany192.box.sample.data.bean.Article;
import com.github.lany192.box.sample.databinding.ItemArticleBinding;
import com.github.lany192.utils.ImageUtils;

import java.util.Date;

public class ArticleBinder extends ItemBinder<Article, ItemArticleBinding> {

    @Override
    public void onClick(@NonNull BindingHolder<ItemArticleBinding> holder, @NonNull View view, Article item, int position) {
        AppRouter.startBrowser(item.getTitle(), item.getLink());
    }

    @Override
    public void convert(@NonNull ItemArticleBinding binding, Article item, int position) {
        ImageUtils.show(binding.image, MockUtils.getImageUrl());
        binding.title.setText(item.getTitle());
        binding.desc.setText(item.getAuthor());
        binding.time.setText(DateUtils.format(new Date(item.getPublishTime())));
    }
}

File: app/avatar/src/main/java/com/lany192/box/avatar/binder/ArticleBinder.kt
package com.lany192.box.avatar.binder

import com.alibaba.android.arouter.launcher.ARouter
import com.github.lany192.arch.items.ItemBinder
import com.github.lany192.arch.utils.DateUtils
import com.github.lany192.extension.load
import com.lany192.box.avatar.MockUtils
import com.lany192.box.avatar.databinding.ItemArticleBinding
import com.lany192.box.network.data.bean.Article
import com.lany192.box.router.provider.BrowserProvider
import java.util.Date
import javax.inject.Inject

class ArticleBinder @Inject constructor() : ItemBinder<Article, ItemArticleBinding>() {
    override fun isFullSpanItem(itemType: Int): Boolean {
        return true
    }

    override fun onItemClick(binding: ItemArticleBinding, item: Article, position: Int) {
        ARouter.getInstance().navigation(BrowserProvider::class.java)
            .startBrowser(item.title, item.link)
    }

    override fun convert(binding: ItemArticleBinding, item: Article, position: Int) {
        binding.image.load(MockUtils.getImageUrl())
        binding.title.text = item.title + position
        binding.desc.text = item.author
        binding.time.text = DateUtils.format(Date(item.publishTime))
    }
}
